http4s/src/main/scala/Http4sConsul.scala

```diff
@@ -62,20 +66,118 @@ final class Http4sConsulClient[F[_]](
   def addCreds(req: Request[F]): Request[F] =
     credentials.fold(req){case (un,pw) => req.putHeaders(Authorization(BasicCredentials(un,pw)))}
 
-  def kvGet(key: Key): F[Option[String]] = {
+  /** A nice place to store the Consul response headers so we can pass them around */
+  case class ConsulHeaders(
+    index:       Long,
+    lastContact: Long,
+    knownLeader: Boolean
+  )
+
+  /** Helper function to get the value of a header out of a Response */
+  def extractHeaderValue(header: String, response: Response[F]): F[String] = {
```

Comment:
Should this return an `F[Option[String]]`, or are we dealing with a very limited set and are willing to call this an error?

Reply:
Really there are only the three headers we need. Could probably just get rid of this function, since it's only used in `extractConsulHeaders` in the final design here, I suppose, if it seems likely to cause confusion.

Reply:
I don't mind exposing it if it's an `Option`, or if it documents the type of error raised on failure.  But yeah, anything additional in the API is a liability, so maybe hiding it is smartest.